aiohttp_client_cache/backends/dynamodb.py

```diff
@@ -1,148 +1,179 @@
+        doc = self._doc(key)
+        table = await self.get_table()
+        await table.delete_item(Key=doc)
+
+    async def read(self, key: str) -> ResponseOrKey:
+        table = await self.get_table()
+        response = await table.get_item(Key=self._doc(key), ProjectionExpression=self.val_attr_name)
+        item = response.get("Item")
+        if item:
+            return pickle.loads(item[self.val_attr_name].value)
+
+    async def write(self, key: str, item: ResponseOrKey) -> None:
+        doc = self._doc(key)
+        doc[self.val_attr_name] = pickle.dumps(item, protocol=pickle.HIGHEST_PROTOCOL)
+
+        # It might be better to incorporate this into is_cacheable, but don't want to serialize more than once
+        # TODO better calc for size that includes whole doc?
+        if len(doc[self.val_attr_name]) > MAX_DOC_SIZE:
+            logger.info(f"Unable to cache large resp")
+            return None
+
+        table = await self.get_table()
+        await table.put_item(Item=doc)
 
     async def clear(self):
-        response = self._scan_table()
-        for v in response['Items']:
-            composite_key = {'namespace': v['namespace'], 'key': v['key']}
-            self._table.delete_item(Key=composite_key)
+        raise NotImplementedError
 
-    # TODO
     async def contains(self, key: str) -> bool:
         raise NotImplementedError
 
-    async def delete(self, key: str):
-        composite_key = {'namespace': self.namespace, 'key': str(key)}
-        response = self._table.delete_item(Key=composite_key, ReturnValues='ALL_OLD')
-        if 'Attributes' not in response:
-            raise KeyError
-
-    # TODO
     async def keys(self) -> Iterable[str]:
         raise NotImplementedError
 
-    async def read(self, key: str) -> ResponseOrKey:
-        response = self._table.get_item(Key={'namespace': self.namespace, 'key': str(key)})
-        return self.unpickle(response.get('Item'))
-
     async def size(self) -> int:
-        expression_attribute_values = {':Namespace': self.namespace}
-        expression_attribute_names = {'#N': 'namespace'}
-        key_condition_expression = '#N = :Namespace'
-        return self._table.query(
-            Select='COUNT',
-            ExpressionAttributeValues=expression_attribute_values,
-            ExpressionAttributeNames=expression_attribute_names,
-            KeyConditionExpression=key_condition_expression,
-        )['Count']
+        raise NotImplementedError
 
     async def values(self) -> Iterable[ResponseOrKey]:
-        response = self._scan_table()
-        return [self.unpickle(item) for item in response.get('Items', [])]
-
-    async def write(self, key: str, item: ResponseOrKey):
-        item_meta = {
-            'namespace': self.namespace,
-            'key': str(key),
-            'value': pickle.dumps(item, protocol=-1),
-        }
-        self._table.put_item(Item=item_meta)
+        raise NotImplementedError
```

Comment:
Could prob impl these for completeness's sake but since many do not seem to be used outside tests I am not sure if they should be a required part of that standard interface?

Reply:
This is use for for getting information from cached responses. For example, something I added to `requests-cache` that I'll likely also add here:

https://github.com/reclosedev/requests-cache/blob/111e03a4a72bc4f20868fa32b63726de84bd6b72/requests_cache/backends/base.py#L35-L40